From 001940573d01d2753e97074d2e8e2526b7099910 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 5 Jan 2024 19:16:26 +0100
Subject: [PATCH 1066/1695] dts: stm32: update remoteproc node compatible

For the STM32MP15 series, the compatible "st,stm32mp1-m4" is not
used in OP-TEE. Only "st,stm32mp1-m4-tee" is used.
With this commit, enabling the remote processor firmware image
authentication will consist of enabling the m4_proc node.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7f95e292a6713c476d03192e261e7dea14dd8385 dts: stm32: update remoteproc node compatible]
Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I064419a28bb5235d0780369a340b7227b3b1cb22
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/401583
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Tested-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp151.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp151.dtsi b/core/arch/arm/dts/stm32mp151.dtsi
index cb69651da..1fc3d973a 100644
--- a/core/arch/arm/dts/stm32mp151.dtsi
+++ b/core/arch/arm/dts/stm32mp151.dtsi
@@ -2068,7 +2068,7 @@
 			     <0x30000000 0x30000000 0x60000>;
 
 		m4_rproc: m4@10000000 {
-			compatible = "st,stm32mp1-m4";
+			compatible = "st,stm32mp1-m4-tee";
 			reg = <0x10000000 0x40000>,
 			      <0x30000000 0x40000>,
 			      <0x38000000 0x10000>;
-- 
2.47.2

